src/test/java/org/folio/rs/service/RetrievalQueueServiceUnitTest.java

```diff
@@ -0,0 +1,187 @@
+import org.folio.rs.domain.dto.MovedEvent;
+import org.folio.rs.domain.dto.MovedEventRequest;
+import org.folio.rs.domain.dto.ResultList;
+import org.folio.rs.domain.dto.User;
+import org.folio.rs.domain.entity.RetrievalQueueRecord;
+import org.folio.rs.mapper.MovedEventMapper;
+import org.folio.rs.repository.RetrievalQueueRepository;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.mockito.junit.jupiter.MockitoSettings;
+import org.mockito.quality.Strictness;
+
+@ExtendWith(MockitoExtension.class)
+@MockitoSettings(strictness = Strictness.LENIENT)
+public class RetrievalQueueServiceUnitTest {
+
+  private static final String ITEM_BARCODE = "653285216743";
+  private static final String PATRON_BARCODE = "543285216734";
+  private static final String CALL_NUMBER = "1234567890";
+  private static final String EFFECTIVE_LOCATION_ID = "53cf956f-c1df-410b-8bea-27f721cca8c0";
+  private static final String REQUESTER_ID = "e546d50a-926a-421f-8400-a041a2e9db79";
+  private static final String HOLD_ID = "7babb1ab-f46f-4c74-8950-bda779440f6f";
+  private static final String PICKUP_LOCATION = "3a40852d-49fd-4df2-a1f9-6e2641a6e91f";
+  private static final String REMOTE_STORAGE_ID = "d3cfdb9e-e364-4209-85e5-8f49fc0969b5";
+  private static final String PATRON_NAME = "John Smith";
+  private static final String INSTANCE_AUTHOR = "Matt Gordon";
+  private static final String STATUS = "Open - Not yet filled";
+  private static final String REQUEST_NOTE = "request note";
+  private static final String INSTANCE_TITLE = "Title";
+  private static final String ITEM_STATUS_NAME = "Paged";
+  private static final String LOCATION_CODE = "KU/CC/DI";
+
+  @InjectMocks
+  private RetrievalQueueService service;
+  @Captor
+  private ArgumentCaptor<RetrievalQueueRecord> captor;
+  @Mock
+  private RetrievalQueueRepository retrievalQueueRepository;
+  @Mock
+  private LocationMappingsService locationMappingsService;
+  @Mock
+  private InventoryClient inventoryClient;
+  @Mock
+  private UsersClient usersClient;
+  @Mock
+  private LocationClient locationClient;
+  @Mock
+  private Item item;
+  @Mock
+  private ResultList<Item> items;
+  @Mock
+  private Contributor contributor;
+  @Mock
+  private LocationMapping locationMapping;
+  @Mock
+  private EffectiveLocation effectiveLocation;
+  @Mock
+  private FolioLocation folioLocation;
+  @Mock
+  private ResultList<User> users;
+  @Mock
+  private User user;
+  @Mock
+  private EffectiveCallNumberComponents callNumberComponents;
+  @Mock
+  private MovedEvent movedEvent;
+  @Mock
+  private MovedEventRequest movedEventRequest;
+  @Mock
+  private MovedEventMapper movedEventMapper;
+
+  @BeforeEach
+  void prepareTestData() {
```

Comment:
This set of mocks doesn't cover any integration and correctness of internal calls by FeignClients implementations: inventoryClient, usersClient, locationCLient. Personally me prefer to use wiremock to have possibility for verification at least of x-okapi-token/x-okapi-tenant. Actually, I don't understand why it is hard to mock just 3 real samples via WireMock instead of huge couple of unclear when(...)

Reply:
We use classic unit tests to check new functionality and mock behavior and responses from inventory, users etc. to check, that our new code working as expected. Personally, for me its harder to read saved json body for wiremock then a couple of when then statements because it contains a lot of unused info for tests . Also its more comfortable for me to change behavior for some test just by changing 1-2 when then statements. If its needed, i can try to redo all tests with wiremock, but it will require some additional capacity